src/lib.rs

```diff
@@ -1,6 +1,10 @@
 #![feature(arbitrary_self_types)]
 #![deny(unsafe_op_in_unsafe_fn)]
 #![allow(dead_code)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
```

Comment:
We can leave this unqualified, and then just use `alloc` instead of `std` everywhere that's relevant (likely just `Box`, `Vec`, and `BTreeSet`).